client/src/components/ItineraryList/ItineraryLegDetails.tsx

```diff
@@ -17,19 +29,41 @@ export function ItineraryLegDetails({ leg, isLast }: { leg: Leg; isLast: boolean
         <b>{leg.mode}</b>{' '}
         {leg.line && (
           <>
-            <u>
+            <a
```

Comment:
I would consider a new component here similar to GraphiQLRouteButton to make this a little bit easier on the eyes.
Something along these lines:
```typescript
<GraphiQLLink title={leg.line?.id} href={...}>
  {leg.line.publicCode} {leg.toEstimatedCall?.destinationDisplay?.frontText}
</GraphiQLLink>
```

Reply:
I would go even further and build the URL in the component.